src/components/reservations/CustomerReservationRow.tsx

```diff
@@ -194,37 +195,65 @@ const CustomerReservationRow = ({ customer, reservation }: IProps): JSX.Element
           </div>
         </div>
       </div>
-      {firstInQueue && (
+      {(firstInQueue || !isCanceled) && (
         <div className={styles.buttons}>
-          <Button
-            variant="secondary"
-            size="small"
-            onClick={() =>
-              dispatch(
-                showOfferModal({
-                  apartment: apartment,
-                  customer: getOfferCustomerData(customer),
-                  isNewOffer: !reservation.offer,
-                  project: project,
-                  reservation: reservation,
-                })
-              )
-            }
-          >
-            {t(`${T_PATH}.offer`)}
-          </Button>
-          {!isInReview && (
-            <>
-              <Button variant="secondary" size="small" onClick={download} disabled={isLoadingContract}>
-                {reservation.project_ownership_type.toLowerCase() === 'haso'
-                  ? t(`${T_PATH}.createContract`)
-                  : t(`${T_PATH}.createDeedOfSale`)}
+          <div>
+            {firstInQueue && (
+              <>
+                <Button
+                  variant="secondary"
+                  size="small"
+                  onClick={() =>
+                    dispatch(
+                      showOfferModal({
+                        apartment: apartment,
+                        customer: mapApartmentReservationCustomerData(customer),
+                        isNewOffer: !reservation.offer,
+                        project: project,
+                        reservation: reservation,
+                      })
+                    )
+                  }
+                >
+                  {t(`${T_PATH}.offer`)}
+                </Button>
+                {!isInReview && (
+                  <>
+                    <Button variant="secondary" size="small" onClick={download} disabled={isLoadingContract}>
+                      {reservation.project_ownership_type.toLowerCase() === 'haso'
+                        ? t(`${T_PATH}.createContract`)
+                        : t(`${T_PATH}.createDeedOfSale`)}
+                    </Button>
+                    <a href={fileUrl} download={fileName} className="hiddenFromScreen" ref={fileRef}>
+                      {t(`${T_PATH}.download`)}
+                    </a>
+                  </>
+                )}
+              </>
+            )}
+          </div>
+          <div>
+            {!isCanceled && (
```

Comment:
If the above comment holds water, then I guess this check is not needed at all.

Reply:
This will be needed as explained in the comment above